src/views/view.c

```diff
@@ -1222,6 +1226,85 @@ int dt_view_image_expose(dt_view_image_over_t *image_over, uint32_t imgid, cairo
   return missing;
 }
 
+void
```

Comment:
@LebedevRI, I have refactored the code, what do you think? I find that the routine is a bit complex now but no strong opinion.